mediapart/ansible-wal-g
# Add downloaded wal-g file check and remove `always` ansible tag

## Description

- Verify wal-g download checksum to avoid installing a corrupted or altered version. Moreover, the unarchiving task is adapted to the fact that the files included in the tar.gz of recent versions of wal-g have changed compared to older versions.
- Remove wal-g `always` ansible tags: we don't want ansible to always execute the wal-g role if we are targeting another role of the playbook.

## Tests
- Change the wal-g version to upgrade or downgrade it while replacing the corresponding sha256 checksum, then validate that the desired version is correctly downloaded and installed.
- Repeat the same test but without updating the checksum and validate that an ansible error is thrown.
- Run ansible on a different tag than the one used to include the wal-g role and validate that the wal-g role is no longer executed.
